analyzer/codechecker_analyzer/arg.py

```diff
@@ -11,6 +11,20 @@
 import argparse
 
 
+class CtuAstModeAction(argparse.Action):
+    """
+    Asserts the presence of ctu mode in the arguments. It is not
+    meaningful to supply --ctu-ast-mode if ctu mode is not enabled.
+    """
+    def __call__(self, parser, namespace, values,
+                 option_string=None):
+        if 'ctu_phases' not in namespace:
+            raise ValueError("Analyzer option 'ctu-ast-mode' requires CTU mode"
+                             " enabled")
```

Comment:
This will look very ugly:
```
Traceback (most recent call last):
  File "~/codechecker/build/CodeChecker/cc_bin/CodeChecker.py", line 117, in main
    args = parser.parse_args()
  File "/usr/lib/python3.6/argparse.py", line 1743, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/lib/python3.6/argparse.py", line 1775, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib/python3.6/argparse.py", line 1963, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/usr/lib/python3.6/argparse.py", line 1940, in consume_positionals
    take_action(action, args)
  File "/usr/lib/python3.6/argparse.py", line 1849, in take_action
    action(self, namespace, argument_values, option_string)
  File "/usr/lib/python3.6/argparse.py", line 1146, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/usr/lib/python3.6/argparse.py", line 1775, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib/python3.6/argparse.py", line 1981, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib/python3.6/argparse.py", line 1921, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib/python3.6/argparse.py", line 1849, in take_action
    action(self, namespace, argument_values, option_string)
  File "~/codechecker/build/CodeChecker/lib/python3/codechecker_analyzer/arg.py", line 22, in __call__
    raise ValueError("Analyzer option 'ctu-ast-mode' requires CTU mode"
ValueError: Analyzer option 'ctu-ast-mode' requires CTU mode enabled
```

So I recommend to do the following:
```suggestion
            LOG.error("Analyzer option 'ctu-ast-mode' requires CTU mode "
                      "enabled!")
            sys.exit(1)
```